doc/acl/Extend-L3V6ACLs.md

```diff
@@ -0,0 +1,428 @@
+     * | MATCH_ICMPV6_TYPE    |      √       |
+     * | MATCH_ICMPV6_CODE    |      √       |
+     * | MATCH_ICMP_TYPE      |      √       |
+     * | MATCH_ICMP_CODE      |      √       |
+     * |-------------------------------------|
+     * | MATCH_IP_PROTOCOL    |      √       |
+     * | MATCH_NEXT_HEADER    |      √       |
+     * | ------------------------------------|
+     * | MATCH_L4_SRC_PORT    |      √       |
+     * | MATCH_L4_DST_PORT    |      √       |
+     * | MATCH_TCP_FLAGS      |      √       |
+     * |-------------------------------------|
+     */
+
+
+
+##### Phase 1:
+On platforms supporting v4 and v6 in a single SAI ACL table, this _new L3V4V6_ ACL table is supported.
+
+##### Phase 2:
+On platforms NOT supporting v4 and v6 in a single SAI ACL table, this new L3V4V6 ACL table will be supported by orchagent's special handling. Orchagent would internally create and manage two SAI ACL tables as shown in the right picture below. The operator is agnostic about the underlying ASIC behavior and only sees a consolidated SONiC ACL table.
+
+
+<p align=center>
+<img src="img/acl-v4v6.png" alt="Hardware optimization for matching v4 in V6 ACL Table optimization">
+</p>
+
+Note: 
+* Due to the lack of SAI APIs to detect this capability, platform checks in orchagent code are used to detect the support.
+* Currently, phase-1 is supported. On non-supporting ASICs, when operator creates an ACL table of type L3V4V6, orchagent throws an error in syslog.
+* Phase-2 will be implemented in the future release.
+  
+##### Pros
+- Readability: L3V4V6 ACL table type conveys to the operator that the ACL table type supports v4 and v6 match qualifiers.
+- Allows the operator the flexibility to use only L3 ACL tables when the deployment does not need v6 rules.
+- Using the same config across ASIC families: Once the future plan of orchagent creating separate v4 and v6 SAI ACL tables on unsupported platform is done, the operator can have the same ACL configuration across ASIC platforms.
+  
+
+#### Implementation
+
+Based on the above design considerations and feedback from the community, **option-C is implemented**.
+New fields are added to the ACL capability in STATE_DB to help applications identify the platforms where the new L3V4V6 ACL table is supported; this is done for both ingress and egress individually.
+
+
+**Matches supported in L3V4V6 table**
+
+- VLAN_ID
+- *IP_TYPE**
+- *ETHER_TYPE**
+- SRC_IPV6
+- DST_IPV6
+- SRC_IP
+- DST_IP
+- ICMPV6_TYPE
+- ICMPV6_CODE
+- ICMP_TYPE
+- ICMP_CODE
+- IP_PROTOCOL
+- NEXT_HEADER
+- L4_SRC_PORT
+- L4_DST_PORT
+- TCP_FLAGS
+
+***Note**: Every ACL Rule, should include at least one of [*IP_TYPE*, *ETHER_TYPE*] in the matching criteria when matching of L3 header fields. This allows orchestrator to decide in which underlying SAI ACL table to place the rule in. If these fields are not provided in the ACL rule, the rule is placed in the SAI IPv4 ACL table and the behavior is dependent on the platform.
```

Comment:
If IP_TYPE, ETHER_TYPE not provided, it is probably useful to apply that rule to both v4 and v6 traffic. eg. if acl-rule is configured only with L4_SRC_PORT/L4_DST_PORT, just one rule can capture both v4/v6 types of traffic. This helps admin to save on configuration lines, and also saves tcam resources by creating only one tcam entry. VLAN_ID, IP_PROTOCOL, L4_SRC_PORT, L4_DST_PORT, TCP_FLAGS are all commonly used fields that can benefit by saving config lines and tcam resources by sharing v4/v6 in one entry.